File: src/main/java/com/baulsupp/oksocial/util/InsecureTrustManager.java
package com.baulsupp.oksocial.util;

import java.security.cert.CertificateException;
import java.security.cert.X509Certificate;
import javax.net.ssl.X509TrustManager;

public class InsecureTrustManager implements X509TrustManager {
  @Override public void checkClientTrusted(X509Certificate[] chain, String authType)
      throws CertificateException {
  }

  @Override public void checkServerTrusted(X509Certificate[] chain, String authType)
      throws CertificateException {
  }

  @Override public X509Certificate[] getAcceptedIssuers() {
    return new X509Certificate[0];
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/security/InsecureTrustManager.kt
package com.baulsupp.oksocial.security

import java.security.cert.X509Certificate
import javax.net.ssl.X509TrustManager

class InsecureTrustManager : X509TrustManager {

  override fun checkClientTrusted(chain: Array<X509Certificate>, authType: String) {
  }

  override fun checkServerTrusted(chain: Array<X509Certificate>, authType: String) {
  }

  override fun getAcceptedIssuers(): Array<X509Certificate> {
    return arrayOf()
  }
}
